### DISCUSSION TITLE
Graphic GameObject - Dashed Line

### DISCUSSION THREAD

**Discussion by 201flaviosilva** [photonstorm/phaser#6115] [Q&A]
Hey there,

Does anyone know if there is any simple way to draw a dashed line in a `Phaser.GameObjects.Graphics`?

- [Here is my project (no dashed) (in Phaser)](https://jsfiddle.net/201flaviosilva/ukj3n5ay/17/);
- [Here is a canvas example](https://jsfiddle.net/201flaviosilva/wq7jv8su/2/) (in canvas, there is [setLineDash](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/setLineDash) function to this effect :))

<img width="326" alt="Screen" src="https://user-images.githubusercontent.com/55815265/168630588-ad9e4731-0130-456f-a324-451b6da42f7c.png">

(Thanks any help :))

**Comment by Wintermilch** [comment]
I have solved it this way:
**distance** is the distance dragged with the mouse and **normalizedDirection** is just the direction the mouse is moving but normalized.
But you can set these values however you want.
`      
        let stepSize = 25;
        let stepAmount = 15;

        let startX = player.x;
        let startY = player.y;
        
        
        for (let i = 0; i < stepAmount; i++) {
            //stepSize = distance / 30;
            stepAmount = Math.ceil(distance / 30);

            this.graphics.lineStyle(3, 0x000, 1);
            if (i % 2 === 0) this.graphics.lineStyle(3, 0x000, 0);


            const targetX = startX - normalizedDirection.x * stepSize;
            const targetY = startY - normalizedDirection.y * stepSize;

            this.graphics.lineBetween(startX, startY, targetX, targetY);

            startX = targetX;
            startY = targetY;
        }`

**Reply by 201flaviosilva** [reply]
Yes, I used this solution more or less too, but I was hoping there would be a more "simpler" solution, but thanks anyway 😅
